surface/src/on_nurbs/fitting_curve_2d.cpp

```diff
@@ -74,7 +74,7 @@ FittingCurve2d::findElement (double xi, const std::vector<double> &elements)
   if (xi >= elements.back ())
     return (int (elements.size ()) - 2);
 
-  for (unsigned i = 0; i < elements.size () - 1; i++)
+  for (unsigned long i = 0; i < elements.size () - 1; i++)
```

Comment:
This looks suspicious. `elements` is a `std::vector` meaning there's no immediate reason for `i` to not be of type `size_t`

Reply:
Ok something is weird. Because this same automatic replacement was done throughout this file. What's the reason for it? Any ideas?

Reply:
Not sure, maybe I read the output wrong ;). I will run check for this file again.

Reply:
```
/media/sf_pcl/surface/src/on_nurbs/fitting_curve_2d.cpp:77:24: warning: loop variable has narrower type 'unsigned int' than iteration's upper bound 'unsigned long' [bugprone-too-small-loop-variable]
  for (unsigned i = 0; i < elements.size () - 1; i++)
                       ^
```
No idea, why clang see `unsigned long` there and not `size_t`

Reply:
I replaced now all `unsigned long` by `size_t`. This causes no new warnings, so clang-tidy seems to be okay with this, too :).